src/AMQPConnection.php

```diff
@@ -99,10 +99,10 @@ public function __construct(string $aliasName, array $connectionDetails = [])
     protected function getConnection(): AbstractConnection
     {
         if (is_null($this->connection)) {
-            if (!isset($this->connection['type'])) {
-                $this->connection['type'] = AMQPStreamConnection::class;
```

Comment:
if the line below $this->createConnectionByType($type); failed for whatever reason, $this->connection remained set as an array with the 'type' key, so calling getConnection() again on the same AMQP connection would return that array instead of going through the if (is_null($this->connection)) branch in this method